src/mbgl/style/layers/location_component_layer.cpp

```diff
@@ -0,0 +1,532 @@
+#include <mbgl/style/conversion/color_ramp_property_value.hpp>
+#include <mbgl/style/conversion/constant.hpp>
+#include <mbgl/style/conversion/json.hpp>
+#include <mbgl/style/conversion/property_value.hpp>
+#include <mbgl/style/conversion/transition_options.hpp>
+#include <mbgl/style/conversion_impl.hpp>
+#include <mbgl/style/layer_observer.hpp>
+#include <mbgl/style/layers/location_component_layer.hpp>
+#include <mbgl/style/layers/location_component_layer_impl.hpp>
+#include <mbgl/util/traits.hpp>
+
+#include <mapbox/eternal.hpp>
+
+namespace mbgl {
+namespace style {
+namespace conversion {
+template <>
+struct Converter<PropertyValue<std::array<double, 3ul>>, void> {
+    optional<PropertyValue<std::array<double, 3ul>>> operator()(const Convertible& value,
+                                                                Error& error,
+                                                                bool,
+                                                                bool) const {
+        optional<std::array<float, 3ul>> a = convert<std::array<float, 3ul>>(value, error);
```

Comment:
how is that helping with precision 😄 ? Maybe keep array of floats?

Reply:
It's only for making it compile. Without this, it will not.

Reply:
proper converter added, resolving this issue.